pkg/consts/views.go

```diff
@@ -97,20 +97,60 @@ function(doc){
       var p = doc.permissions[k];
       var selector = p.selector || "_id";
       for (var i=0; i<p.values.length; i++) {
-        emit([p.type, selector, p.values[i]], p.verbs);
+				emit([p.type, selector, p.values[i]], p.verbs);
       }
     });
   }
 }`,
 }
 
+// SharedWithMePermissionsView returns the list of permissions associated with
+// sharings and for which the user is a recipient.
+var SharedWithMePermissionsView = &couchdb.View{
+	Name:    "sharedWithMePermissions",
+	Doctype: Sharings,
+	Map: `
+function(doc) {
+	if (!doc.owner && doc.permissions) {
+		Object.keys(doc.permissions).forEach(function(k) {
+			var rule = doc.permissions[k];
+			var selector = rule.selector || "_id";
+			for (var i=0; i<rule.values.length; i++) {
+				emit([rule.type], rule);
+			}
+		});
+	}
+}`,
+}
+
+// SharedWithOthersPermissionsView returns the list of permissions associated
+// with sharings and for which the user is the sharer.
+var SharedWithOthersPermissionsView = &couchdb.View{
+	Name:    "sharedWithOthersPermissions",
+	Doctype: Sharings,
+	Map: `
+function(doc) {
+	if (doc.owner && doc.permissions) {
+		Object.keys(doc.permissions).forEach(function(k) {
+			var rule = doc.permissions[k];
+			var selector = rule.selector || "_id";
+			for (var i=0; i<rule.values.length; i++) {
+				emit([rule.type], rule);
+			}
+		});
+	}
+}`,
```

Comment:
Maybe you can merge `SharedWithMePermissionsView` and `SharedWithOthersPermissionsView` in a single view:

```js
emit([!doc.owner, rule.type], rule);
```